astrovascpy/bloodflow.py

```diff
@@ -731,15 +748,22 @@ def _solve_linear(laplacian, input_flow):
 
     petsc_res_norm = None
     if MPI_RANK == 0:
-        petsc_res_norm = np.linalg.norm(input_flow - laplacian * result_petsc_sp)
-    PETSc.Sys.Print(f"-> PETSc residual norm = {petsc_res_norm}")
+        if petsc_solver.getIterationNumber() == petsc_solver.max_it:
+            L.warning(f"Reached maximum number of iteration {petsc_solver.max_it}.")
 
     if bool(int(os.getenv("DEBUG_BFS", "0"))) and MPI_RANK == 0:
-        scipy_res_norm = np.linalg.norm(input_flow - laplacian * result)
-        print(f"-> SciPy residual norm = {scipy_res_norm}")
-        assert np.allclose(
-            petsc_res_norm, scipy_res_norm, rtol=1e-6, atol=1e-6
-        ), "PETSc gives different solution compared to SciPy!"
+        petsc_res_norm = np.linalg.norm(input_flow - laplacian * result_petsc_sp)  # l2 norm
+        input_flow_norm = np.linalg.norm(input_flow)
+        PETSc.Sys.Print("Laplacian matrix size: ", np.shape(laplacian))
```

Comment:
Won't it be too verbose? I would emit the message only if the log level is equal to INFO or above.

Reply:
This part of the code runs only if the environment variable `DEBUG_BFS` is set to 1.  By default, it is 0.  
It servers only for debugging purposes. The user should not use it. 
Actually the purpose of `DEBUG_BFS` is only to print this information.  Should we keep this approach? (for me it is ok)

Reply:
Sorry, I didn't look at the code thoroughly enough. It is fine like this.